src/main/java/com/conveyal/datatools/manager/jobs/PeliasUpdateJob.java

```diff
@@ -0,0 +1,240 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+
+import static com.mongodb.client.model.Filters.in;
+
+public class PeliasUpdateJob extends MonitorableJob {
+    /**
+     * The deployment to send to Pelias
+     */
+    private final Deployment deployment;
+
+    /**
+     * The workerId our request has on the webhook server. Used to get status updates
+     */
+    private String workerId;
+
+
+    /**
+     * Timer used to poll the status endpoint
+     */
+    private final Timer timer;
+
+    /**
+     * The number of webhook status requests allowed to fail before considering the server down
+     */
+    private int webhookStatusFailuresAllowed = 3;
+
+    /**
+     * S3 URI to upload logs to
+     */
+    private final AmazonS3URI logUploadS3URI;
+
+    public PeliasUpdateJob(Auth0UserProfile owner, String name, Deployment deployment, AmazonS3URI logUploadS3URI) {
+        super(owner, name, JobType.UPDATE_PELIAS);
+        this.deployment = deployment;
+        this.timer = new Timer();
+        this.logUploadS3URI = logUploadS3URI;
+    }
+
+    /**
+     * This method must be overridden by subclasses to perform the core steps of the job.
+     */
+    @Override
+    public void jobLogic() throws Exception {
+        status.message = "Launching custom geocoder update request";
+        workerId = this.makeWebhookRequest();
+        status.percentComplete = 1.0;
+
+        // Give server 1 second to create worker
+        Thread.sleep(1000);
+        // Check status every 2 seconds
+        timer.schedule(new StatusChecker(), 0, 2000);
+
+    }
+
+    private void getWebhookStatus() {
+        URI url = getWebhookURI(deployment.peliasWebhookUrl + "/status/" + workerId);
+
+        SimpleHttpResponse response = HttpUtils.httpRequestRawResponse(url, 1000, HttpUtils.REQUEST_METHOD.GET, null);
+
+        // Convert raw body to JSON
+        String jsonResponse;
+        try {
+            jsonResponse = response.body;
+        }
+        catch (NullPointerException ex) {
+            if (--webhookStatusFailuresAllowed == 0) {
+                status.fail("Webhook status did not provide a response!", ex);
+                timer.cancel();
+            }
+            return;
+        }
+
+        // Parse JSON
+        PeliasWebhookStatusMessage statusResponse;
+        try {
+            statusResponse = JsonUtil.objectMapper.readValue(jsonResponse, PeliasWebhookStatusMessage.class);
+        } catch (IOException ex) {
+            status.fail("Server refused to provide a valid status update. Are the credentials correct?", ex);
+            timer.cancel();
+            return;
+        }
```

Comment:
Here is a suggestion for being a little more tolerant of various errors that could occur such that even RuntimeExceptions (non-checked exceptions) get caught.

```suggestion
        PeliasWebhookStatusMessage statusResponse;
        
        try {
            SimpleHttpResponse response = HttpUtils.httpRequestRawResponse(url, 1000, HttpUtils.REQUEST_METHOD.GET, null);

            // Convert raw body to PeliasWebhookStatusMessage
            String jsonResponse = response.body;
            statusResponse = JsonUtil.objectMapper.readValue(jsonResponse, PeliasWebhookStatusMessage.class);
        } catch (Exception ex) {
            if (--webhookStatusFailuresAllowed == 0) {
                status.fail("Webhook status did not provide a valid response!", ex);
                timer.cancel();
            }
            return;
        }
```

Reply:
This catches more and is cleaner thanks! Refactored in 0ddf61f8d6e3a96db61d37545bebed5dfc25445f